### DISCUSSION TITLE
Setting up kibot in a ubuntu 22 environment with kicad 7.0.1/7.0.2

### DISCUSSION THREAD

**Discussion by RensSluimer** [INTI-CMNB/KiBot#426] [General]
As i was looking into using kibot in a CI/CD non github environment.  I wanted to try kibot first out in a ubuntu environment. As i was looking into the installation process. I found out that even with the 1.6.1-3 version of kibot. The quick-start command would give an error on the project files that are made in kicad 7.0.1/7.0.2. As it had installed a 6.0.x verions of kicad. Which can't work with kicad 7.0.x project files.
After some trail and error i found out that if you add the kicad7.0 repo first to your linux environment. The install of kibot would automatically grab the new version.
I did the following to get kibot working on a ubuntu environment, without having the manually install all the extensions and such.
```
  sudo add-apt-repository ppa:kicad/kicad-7.0-releases
  wget https://set-soft.github.io/debian/kibot.list
  sudo cp kibot.list /etc/apt/sources.list.d/
  wget https://set-soft.github.io/debian/kibot.gpg
  sudo cp kibot.gpg /etc/apt/
  sudo apt-get update
  sudo apt-get install kibot
```
It is almost the same as the steps in [repo](https://set-soft.github.io/debian/) with the exception of first adding the kicad repo to your environment.

**Comment by set-soft** [comment]
Hi @RensSluimer !

> As it had installed a 6.0.x verions of kicad.

This has nothing to do with KiBot, is your package manager which chooses the KiCad version. KiBot just depends on *kicad*.

If you install KiBot in a system without KiCad installed the package manager will look for KiCad. KiCad 7 was released in 2023, so Ubuntu 22 (2022 Ubuntu) will use the ofitial KiCad for Ubuntu 22 (6.0.x).

As you already found, in order to get KiCad 7 you *must* tell to Ubuntu that KiCad 7 exists. This is what you do adding the PPA.

KiBot supports KiCad 5, 6, 7 and maybe 7.99 (8 WIP). There is no way for KiBot to force KiCad 7 by default.

That said, if you are looking for using KiBot in a CI/CD environment I'll suggest using already existent docker images. If you need to add tools to them just use them as base (FROM) in your docker files. It will be much easier. 

You can get them from the [kicad_auto_test v7](https://github.com/INTI-CMNB/kicad_auto_test/pkgs/container/kicad7_auto_full) package. You can use *1.6.1* tag for the current stable, or *dev* tag for the current development (1.6.2).  Or you can look in the [kicad_auto_test general](https://github.com/INTI-CMNB/kicad_auto_test/pkgs/container/kicad_auto_test) package, you'll find *ki7* tag, which points to the last stable KiBot.

These images contains KiBot and all its dependencies, even the suggested ones, like Blender. The base OS is Debian 11, which is the base for Ubuntu 20/21 (aprox.).

**Reply by RensSluimer** [reply]
Ah yes, that would explain why it was grabbing the 6.0.x versions 

I was just trying to add an explanation for people who also try to install it. And encounter the same problem. As i didn't read it somewhere. So i thought maybe specify it for other people.
But your explanation is way more extended and detailed 

I was already looking into the docker images. They are really useful for that, yes.
But i wanted to explore and experiment a bit with kibot. That's why i wanted to install it on a ubuntu environment 

Thank you for the extended information. I will continue exploring all the functionalities
